### PR TITLE
fix(core): align HighlightsClient with the live highlights API contract

### PR DESCRIPTION
**Pull request by Dustin-Kelley**
## Summary

`HighlightsClient` was written against an outdated version of the highlights API and has never worked against the live endpoints (its integration tests were `describe.skip`'d from day one). Every request was rejected, which surfaced as highlights silently failing to fetch/create/delete in the `BibleReader` wiring work (YPE-1034).

Verified against the current [API reference](https://developers.youversion.com/api/highlights); the live endpoint responds `401 "Missing or invalid Bearer token"` to the old request shape.

Changes:

- **Auth**: send the OAuth token as an `Authorization: Bearer <token>` header instead of a `lat` query parameter
- **Naming**: use the API's `bible_id` field on the wire for GET/POST/DELETE; the SDK keeps `version_id` in its public types (consistent with the rest of the SDK) and maps at the boundary
- **Create envelope**: POST now sends the required `{ request_id: <uuid>, highlight: { bible_id, passage_id, color } }` body for idempotent retries
- **Required params**: `getHighlights` now requires `version_id` + `passage_id` (verse or chapter USFM) and `deleteHighlight` requires `version_id`, matching the API's required parameters; `useHighlights` signatures updated accordingly
- **Validation**: responses are parsed with Zod (wire schema) and mapped to the SDK shape
- **Tests**: un-skipped and rewrote `highlights.test.ts` (10 tests); the shared MSW handlers now *enforce* the documented contract (401 without Bearer, 400 on wrong params/body) so future drift fails tests

## Notes for reviewers

- Consumers who signed in before requesting the `highlights` permission need to re-authenticate to get a token with `read_highlights`/`write_highlights` scopes.
- Pre-existing failures not touched by this PR: 4 hooks tests (`YouVersionProvider.test.tsx`, missing localStorage in jsdom env) and 5 UI tests (`bible-reader.test.tsx` theme settings) fail identically on `main`.

## Test plan

- [x] `pnpm --filter @youversion/platform-core test` — 308 passed, including 10 rewritten highlights contract tests
- [x] `pnpm --filter @youversion/platform-react-hooks test` — only the 4 pre-existing `YouVersionProvider` failures remain (identical on main)
- [x] `pnpm typecheck` / `pnpm lint` — clean
- [ ] Manual verification against the live API with a signed-in session that has the `highlights` scope (blocked on a real token; follow-up in YPE-1034 wiring)

Made with [Cursor](https://cursor.com)

<h3>Greptile Summary</h3>

This PR fixes `HighlightsClient` to match the live YouVersion highlights API contract. Every request was previously rejected with 401 because auth was sent as a `lat` query parameter instead of an `Authorization: Bearer` header, and field names / request shapes differed from the documented API. The rewrite also un-skips and rewrites the highlights test suite so the MSW handlers now enforce the contract (auth, field names, enveloped POST body), preventing future drift.

- **Auth & field mapping**: Bearer header replaces `lat` query param; `bible_id` is used on the wire while the SDK keeps `version_id` in public types, with a clean `toHighlight()` mapper at the boundary.
- **Request / response hardening**: POST sends the required `{ request_id, highlight: { … } }` envelope; `getHighlights` and `deleteHighlight` now enforce their required parameters; Zod schemas validate all responses; 204 (no highlights) is treated as an empty collection; uppercase hex colors are normalized to lowercase before sending.
- **New surface**: `getRecentColors()` added to `HighlightsClient` and exposed from `useHighlights` for color-picker UIs.

<h3>Confidence Score: 5/5</h3>

Safe to merge — all four endpoint contract fixes are correctly implemented and the MSW handlers now enforce them so any future drift will fail CI.

The implementation changes are precise and internally consistent: Bearer auth, wire-field mapping, enveloped POST body, 204 handling, and Zod validation all align with the documented API. The HEX_COLOR_REGEX was updated to case-insensitive, addressing the mismatch flagged in the previous review cycle. Tests cover the happy path, 204 edge case, malformed-response error, and uppercase color normalization. The changeset bump level is worth a second look but does not affect runtime correctness.

.changeset/fix-highlights-api-contract.md — worth a second look on the minor-vs-major bump decision for 2.x packages.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| packages/core/src/highlights.ts | Rewrites HighlightsClient to use Bearer auth, bible_id field mapping, enveloped POST body, Zod response validation, and 204 handling; adds getRecentColors(). All contract fixes are correct and well-commented. |
| packages/core/src/schemas/highlight.ts | Adds wire schemas (HighlightWireSchema, HighlightCollectionWireSchema, RecentHighlightColorsWireSchema) and toHighlight() mapper. HEX_COLOR_REGEX is now case-insensitive, fixing the previously flagged mismatch. |
| packages/core/src/client.ts | Adds optional headers parameter to post() and delete(), enabling callers to inject Authorization: Bearer headers without touching the shared defaultHeaders. |
| packages/hooks/src/useHighlights.ts | Makes options and deleteOptions required to match the now-required API params; adds getRecentColors() callback; dep-array comment explains the primitive-keying strategy. |
| packages/core/src/__tests__/handlers.ts | MSW handlers now enforce Bearer auth (401 on missing token), bible_id naming, and enveloped POST body (400 on wrong shape), so client drift breaks tests immediately. |
| packages/core/src/__tests__/highlights.test.ts | Un-skipped and rewrote 10 tests covering the new contract; tests the Bearer header, bible_id param, enveloped POST, 204 handling, malformed-response error, and uppercase color normalization. |
| packages/hooks/src/useHighlights.test.tsx | Test suite updated to match the new required-options signatures and adds coverage for getRecentColors delegation. |
| .changeset/fix-highlights-api-contract.md | Changeset marked as minor for both 2.x packages; the required-param promotions are technically semver-major, though the feature was never functional before. |

<h3>Sequence Diagram</h3>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
sequenceDiagram
    participant App as App / useHighlights
    participant HC as HighlightsClient
    participant AC as ApiClient
    participant API as Highlights API

    Note over HC: getAuthToken(lat?) → Bearer header

    App->>HC: "getHighlights({ version_id, passage_id }, lat?)"
    HC->>HC: validateVersionId / validatePassageId
    HC->>AC: "get("/v1/highlights", { bible_id, passage_id }, { Authorization: Bearer })"
    AC->>API: "GET /v1/highlights?bible_id=&passage_id= + Bearer"
    API-->>AC: "200 { data: [{ bible_id, passage_id, color }] } OR 204"
    AC-->>HC: raw response
    HC->>HC: HighlightCollectionWireSchema.safeParse()
    HC->>HC: data.map(toHighlight) — bible_id → version_id
    HC-->>App: "Collection<Highlight>"

    App->>HC: "createHighlight({ version_id, passage_id, color }, lat?)"
    HC->>HC: validate + color.toLowerCase()
    HC->>AC: "post("/v1/highlights", { request_id, highlight: { bible_id, passage_id, color } }, headers)"
    AC->>API: POST /v1/highlights + Bearer
    API-->>AC: "201 { bible_id, passage_id, color }"
    AC-->>HC: raw response
    HC->>HC: HighlightWireSchema.safeParse() → toHighlight()
    HC-->>App: Highlight

    App->>HC: "deleteHighlight(passageId, { version_id }, lat?)"
    HC->>HC: validatePassageId / validateVersionId
    HC->>AC: "delete("/v1/highlights/{passageId}", { bible_id }, { Authorization: Bearer })"
    AC->>API: "DELETE /v1/highlights/{passageId}?bible_id= + Bearer"
    API-->>AC: 204
    HC-->>App: void

    App->>HC: getRecentColors(lat?)
    HC->>AC: "get("/v1/highlights/recent-colors", undefined, { Authorization: Bearer })"
    AC->>API: GET /v1/highlights/recent-colors + Bearer
    API-->>AC: "200 { data: [{ color }] } OR 204"
    HC->>HC: RecentHighlightColorsWireSchema.safeParse()
    HC-->>App: string[]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
sequenceDiagram
    participant App as App / useHighlights
    participant HC as HighlightsClient
    participant AC as ApiClient
    participant API as Highlights API

    Note over HC: getAuthToken(lat?) → Bearer header

    App->>HC: "getHighlights({ version_id, passage_id }, lat?)"
    HC->>HC: validateVersionId / validatePassageId
    HC->>AC: "get("/v1/highlights", { bible_id, passage_id }, { Authorization: Bearer })"
    AC->>API: "GET /v1/highlights?bible_id=&passage_id= + Bearer"
    API-->>AC: "200 { data: [{ bible_id, passage_id, color }] } OR 204"
    AC-->>HC: raw response
    HC->>HC: HighlightCollectionWireSchema.safeParse()
    HC->>HC: data.map(toHighlight) — bible_id → version_id
    HC-->>App: "Collection<Highlight>"

    App->>HC: "createHighlight({ version_id, passage_id, color }, lat?)"
    HC->>HC: validate + color.toLowerCase()
    HC->>AC: "post("/v1/highlights", { request_id, highlight: { bible_id, passage_id, color } }, headers)"
    AC->>API: POST /v1/highlights + Bearer
    API-->>AC: "201 { bible_id, passage_id, color }"
    AC-->>HC: raw response
    HC->>HC: HighlightWireSchema.safeParse() → toHighlight()
    HC-->>App: Highlight

    App->>HC: "deleteHighlight(passageId, { version_id }, lat?)"
    HC->>HC: validatePassageId / validateVersionId
    HC->>AC: "delete("/v1/highlights/{passageId}", { bible_id }, { Authorization: Bearer })"
    AC->>API: "DELETE /v1/highlights/{passageId}?bible_id= + Bearer"
    API-->>AC: 204
    HC-->>App: void

    App->>HC: getRecentColors(lat?)
    HC->>AC: "get("/v1/highlights/recent-colors", undefined, { Authorization: Bearer })"
    AC->>API: GET /v1/highlights/recent-colors + Bearer
    API-->>AC: "200 { data: [{ color }] } OR 204"
    HC->>HC: RecentHighlightColorsWireSchema.safeParse()
    HC-->>App: string[]
```

</a>

<details open><summary><h3>Comments Outside Diff (2)</h3></summary>

1. `packages/core/src/highlights.ts`, line 59-71 ([link](https://github.com/youversion/platform-sdk-react/blob/906379b87a9cedc83521fd7a3cab698f691628bb/packages/core/src/highlights.ts#L59-L71)) 

   <a href="#">P2</a> **`localStorage` guard makes the auth fallback silently different across environments.** When `typeof localStorage === 'undefined'` (Node.js test environment, SSR), `getAuthToken(undefined)` returns `null` and immediately throws. That's the intended path for unauthenticated callers. However, the guard also means that any call inside a server-side context that *should* find a token in `YouVersionPlatformConfiguration` will always throw, even if the platform configuration has been populated another way. Worth a comment that makes this intent explicit, and a test confirming SSR callers must pass `lat` explicitly.

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: packages/core/src/highlights.ts
   Line: 59-71

   Comment:
   **`localStorage` guard makes the auth fallback silently different across environments.** When `typeof localStorage === 'undefined'` (Node.js test environment, SSR), `getAuthToken(undefined)` returns `null` and immediately throws. That's the intended path for unauthenticated callers. However, the guard also means that any call inside a server-side context that *should* find a token in `YouVersionPlatformConfiguration` will always throw, even if the platform configuration has been populated another way. Worth a comment that makes this intent explicit, and a test confirming SSR callers must pass `lat` explicitly.

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

   Note: If this suggestion doesn't match your team's coding style, reply to this and let me know. I'll remember it for next time!

   <a href="https://app.greptile.com/ide/claude-code?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fcore%2Fsrc%2Fhighlights.ts%0ALine%3A%2059-71%0A%0AComment%3A%0A**%60localStorage%60%20guard%20makes%20the%20auth%20fallback%20silently%20different%20across%20environments.**%20When%20%60typeof%20localStorage%20%3D%3D%3D%20'undefined'%60%20%28Node.js%20test%20environment%2C%20SSR%29%2C%20%60getAuthToken%28undefined%29%60%20returns%20%60null%60%20and%20immediately%20throws.%20That's%20the%20intended%20path%20for%20unauthenticated%20callers.%20However%2C%20the%20guard%20also%20means%20that%20any%20call%20inside%20a%20server-side%20context%20that%20*should*%20find%20a%20token%20in%20%60YouVersionPlatformConfiguration%60%20will%20always%20throw%2C%20even%20if%20the%20platform%20configuration%20has%20been%20populated%20another%20way.%20Worth%20a%20comment%20that%20makes%20this%20intent%20explicit%2C%20and%20a%20test%20confirming%20SSR%20callers%20must%20pass%20%60lat%60%20explicitly.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=youversion%2Fplatform-sdk-react&pr=279&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaude.svg?v=6">Fix in Claude Code</picture></a> <a href="https://app.greptile.com/api/ide/cursor?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fcore%2Fsrc%2Fhighlights.ts%0ALine%3A%2059-71%0A%0AComment%3A%0A**%60localStorage%60%20guard%20makes%20the%20auth%20fallback%20silently%20different%20across%20environments.**%20When%20%60typeof%20localStorage%20%3D%3D%3D%20'undefined'%60%20%28Node.js%20test%20environment%2C%20SSR%29%2C%20%60getAuthToken%28undefined%29%60%20returns%20%60null%60%20and%20immediately%20throws.%20That's%20the%20intended%20path%20for%20unauthenticated%20callers.%20However%2C%20the%20guard%20also%20means%20that%20any%20call%20inside%20a%20server-side%20context%20that%20*should*%20find%20a%20token%20in%20%60YouVersionPlatformConfiguration%60%20will%20always%20throw%2C%20even%20if%20the%20platform%20configuration%20has%20been%20populated%20another%20way.%20Worth%20a%20comment%20that%20makes%20this%20intent%20explicit%2C%20and%20a%20test%20confirming%20SSR%20callers%20must%20pass%20%60lat%60%20explicitly.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&pr=279&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursorDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursor.svg?v=6">Fix in Cursor</picture></a> <a href="https://app.greptile.com/api/ide/codex?prompt=IMPORTANT%3A%20Work%20in%20the%20repository%20%22youversion%2Fplatform-sdk-react%22%20on%20the%20existing%20branch%20%22dk%2Ffix-highlights-api-contract%22.%20Checkout%20that%20branch%20%E2%80%94%20do%20NOT%20create%20a%20new%20branch%20or%20open%20a%20new%20PR.%20Push%20your%20changes%20to%20%22dk%2Ffix-highlights-api-contract%22.%0A%0AThis%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fcore%2Fsrc%2Fhighlights.ts%0ALine%3A%2059-71%0A%0AComment%3A%0A**%60localStorage%60%20guard%20makes%20the%20auth%20fallback%20silently%20different%20across%20environments.**%20When%20%60typeof%20localStorage%20%3D%3D%3D%20'undefined'%60%20%28Node.js%20test%20environment%2C%20SSR%29%2C%20%60getAuthToken%28undefined%29%60%20returns%20%60null%60%20and%20immediately%20throws.%20That's%20the%20intended%20path%20for%20unauthenticated%20callers.%20However%2C%20the%20guard%20also%20means%20that%20any%20call%20inside%20a%20server-side%20context%20that%20*should*%20find%20a%20token%20in%20%60YouVersionPlatformConfiguration%60%20will%20always%20throw%2C%20even%20if%20the%20platform%20configuration%20has%20been%20populated%20another%20way.%20Worth%20a%20comment%20that%20makes%20this%20intent%20explicit%2C%20and%20a%20test%20confirming%20SSR%20callers%20must%20pass%20%60lat%60%20explicitly.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=youversion%2Fplatform-sdk-react&pr=279&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCodexDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCodex.svg?v=6">Fix in Codex</picture></a>

2. `packages/hooks/src/useHighlights.ts`, line 45-51 ([link](https://github.com/youversion/platform-sdk-react/blob/906379b87a9cedc83521fd7a3cab698f691628bb/packages/hooks/src/useHighlights.ts#L45-L51)) 

   <a href="#">P2</a> **`options` object identity is unstable if passed as an inline literal.** The factory `() => highlightsClient.getHighlights(options)` captures the full `options` reference, but the dependency array correctly uses only the primitives `options.version_id` and `options.passage_id`. This means the closure may hold a stale object reference while the primitives stay stable — in practice this is fine here, but callers who add fields to `GetHighlightsOptions` later might be surprised. A quick clarifying comment on the dep-array strategy would help future maintainers.

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: packages/hooks/src/useHighlights.ts
   Line: 45-51

   Comment:
   **`options` object identity is unstable if passed as an inline literal.** The factory `() => highlightsClient.getHighlights(options)` captures the full `options` reference, but the dependency array correctly uses only the primitives `options.version_id` and `options.passage_id`. This means the closure may hold a stale object reference while the primitives stay stable — in practice this is fine here, but callers who add fields to `GetHighlightsOptions` later might be surprised. A quick clarifying comment on the dep-array strategy would help future maintainers.

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

   Note: If this suggestion doesn't match your team's coding style, reply to this and let me know. I'll remember it for next time!

   <a href="https://app.greptile.com/ide/claude-code?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fhooks%2Fsrc%2FuseHighlights.ts%0ALine%3A%2045-51%0A%0AComment%3A%0A**%60options%60%20object%20identity%20is%20unstable%20if%20passed%20as%20an%20inline%20literal.**%20The%20factory%20%60%28%29%20%3D%3E%20highlightsClient.getHighlights%28options%29%60%20captures%20the%20full%20%60options%60%20reference%2C%20but%20the%20dependency%20array%20correctly%20uses%20only%20the%20primitives%20%60options.version_id%60%20and%20%60options.passage_id%60.%20This%20means%20the%20closure%20may%20hold%20a%20stale%20object%20reference%20while%20the%20primitives%20stay%20stable%20%E2%80%94%20in%20practice%20this%20is%20fine%20here%2C%20but%20callers%20who%20add%20fields%20to%20%60GetHighlightsOptions%60%20later%20might%20be%20surprised.%20A%20quick%20clarifying%20comment%20on%20the%20dep-array%20strategy%20would%20help%20future%20maintainers.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=youversion%2Fplatform-sdk-react&pr=279&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaude.svg?v=6">Fix in Claude Code</picture></a> <a href="https://app.greptile.com/api/ide/cursor?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fhooks%2Fsrc%2FuseHighlights.ts%0ALine%3A%2045-51%0A%0AComment%3A%0A**%60options%60%20object%20identity%20is%20unstable%20if%20passed%20as%20an%20inline%20literal.**%20The%20factory%20%60%28%29%20%3D%3E%20highlightsClient.getHighlights%28options%29%60%20captures%20the%20full%20%60options%60%20reference%2C%20but%20the%20dependency%20array%20correctly%20uses%20only%20the%20primitives%20%60options.version_id%60%20and%20%60options.passage_id%60.%20This%20means%20the%20closure%20may%20hold%20a%20stale%20object%20reference%20while%20the%20primitives%20stay%20stable%20%E2%80%94%20in%20practice%20this%20is%20fine%20here%2C%20but%20callers%20who%20add%20fields%20to%20%60GetHighlightsOptions%60%20later%20might%20be%20surprised.%20A%20quick%20clarifying%20comment%20on%20the%20dep-array%20strategy%20would%20help%20future%20maintainers.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&pr=279&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursorDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursor.svg?v=6">Fix in Cursor</picture></a> <a href="https://app.greptile.com/api/ide/codex?prompt=IMPORTANT%3A%20Work%20in%20the%20repository%20%22youversion%2Fplatform-sdk-react%22%20on%20the%20existing%20branch%20%22dk%2Ffix-highlights-api-contract%22.%20Checkout%20that%20branch%20%E2%80%94%20do%20NOT%20create%20a%20new%20branch%20or%20open%20a%20new%20PR.%20Push%20your%20changes%20to%20%22dk%2Ffix-highlights-api-contract%22.%0A%0AThis%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fhooks%2Fsrc%2FuseHighlights.ts%0ALine%3A%2045-51%0A%0AComment%3A%0A**%60options%60%20object%20identity%20is%20unstable%20if%20passed%20as%20an%20inline%20literal.**%20The%20factory%20%60%28%29%20%3D%3E%20highlightsClient.getHighlights%28options%29%60%20captures%20the%20full%20%60options%60%20reference%2C%20but%20the%20dependency%20array%20correctly%20uses%20only%20the%20primitives%20%60options.version_id%60%20and%20%60options.passage_id%60.%20This%20means%20the%20closure%20may%20hold%20a%20stale%20object%20reference%20while%20the%20primitives%20stay%20stable%20%E2%80%94%20in%20practice%20this%20is%20fine%20here%2C%20but%20callers%20who%20add%20fields%20to%20%60GetHighlightsOptions%60%20later%20might%20be%20surprised.%20A%20quick%20clarifying%20comment%20on%20the%20dep-array%20strategy%20would%20help%20future%20maintainers.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=youversion%2Fplatform-sdk-react&pr=279&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCodexDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCodex.svg?v=6">Fix in Codex</picture></a>
</details>

<sub>Reviews (5): Last reviewed commit: ["docs(highlights): clarify auth fallback ..."](https://github.com/youversion/platform-sdk-react/commit/cef7b5b53943a75ef56b96ad22878b04074190bf) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=42792361)</sub>

### CODE REVIEW

File: packages/core/src/schemas/highlight.ts [hunk 1]
```diff
@@ -1,23 +1,69 @@
 import { z } from 'zod';
 
-const _HighlightSchema = z.object({
+// Case-insensitive: the API may echo colors in any case, and the client's input
+// validator accepts both, so response parsing must not reject uppercase hex.
+const HEX_COLOR_REGEX = /^[0-9a-f]{6}$/i;
+
```
Comment (cameronapak) [inline packages/core/src/schemas/highlight.ts:6]: gg

File: packages/core/src/schemas/highlight.ts [hunk 2]
```diff
@@ -1,23 +1,69 @@
...
+/**
+ * Wire format used by the highlights API (`/v1/highlights`), which names the
+ * Bible version field `bible_id`. The SDK exposes it as `version_id` for
+ * consistency with the rest of the public API (see {@link toHighlight}).
```
Comment (cameronapak) [inline packages/core/src/schemas/highlight.ts:10]: question: what is this link ?

**Review by cameronapak — DISMISSED** [review]

**Comment by Dustin-Kelley** [comment]
@greptile review

**Review by cameronapak — APPROVED** [review]
